Merge/foundation release/1.13.4 to 1.13.1

ziogaschr opened the pull request.

karalabe committed:
params: begin Geth v1.13.2 release cycle

fjl committed:
cmd/evm: fix some issues with the evm run command (#28109)

* cmd/evm: improve flags handling

This fixes some issues with flags in cmd/evm. The supported flags did not
actually show up in help output because they weren't categorized. I'm also
adding the VM-related flags to the run command here so they can be given
after the subcommand name. So it can be run like this now:

   ./evm run --code 6001 --debug

* cmd/evm: enable all forks by default in run command

The default genesis was just empty with no forks at all, which is annoying because
contracts will be relying on opcodes introduced in a fork. So this changes the default to
have all forks enabled.

* core/asm: fix some issues in the assembler

This fixes minor bugs in the old assembler:

- It is now possible to have comments on the same line as an instruction.
- Errors for invalid numbers in the jump instruction are reported better
- Line numbers in errors were off by one

jsvisa committed:
core/rawdb: fix typo in comment (#28140)

bnovil committed:
eth: fix typo in comment (#28146)

jsvisa committed:
cmd/devp2p: use bootnodes as crawl input (#28139)

This PR makes the tool use the --bootnodes list as the input to devp2p crawl.
The flag will take effect if the input/output.json file is missing or empty.

holiman committed:
go.mod: use existing version of karalabe/usb (#28127)

There is no 0.0.3 release of karalabe/usb.

gballet committed:
trie: add getter for preimage store in trie.Database (#28155)

phenix3443 committed:
cmd/utils: fix typo in comment (#28159)

jsvisa committed:
cmd/clef: suppress fsnotify error if keydir not exists (#28160)

As the keydir will be automatically created after an account is created, no error message if the watcher is failed.

jsvisa committed:
core/rawdb: no need to run truncateFile for readonly mode (#28145)

Avoid truncating files, if ancients are opened in readonly mode. With this change, we return error instead of trying (and failing)  to repair

rjl493456442 committed:
trie: remove internal nodes between shortNode and child in path mode (#28163)

* trie: remove internal nodes between shortNode and child in path mode

* trie: address comments

* core/rawdb, trie: address comments

* core/rawdb: delete unused func

* trie: change comments

* trie: add missing tests

* trie: fix lint

rjl493456442 committed:
trie/triedb/pathdb: improve error log (#28177)

karalabe committed:
core/state/snapshot: be very noisy if the generator hits a trie error (#28178)

karalabe committed:
cmd/geth: print progress logs when iterating large contracts too (#28179)

jsvisa committed:
core/rawdb: use readonly file lock in readonly mode (#28180)

This allows using the freezer from multiple processes at once
in read-only mode.

Co-authored-by: Martin Holst Swende <martin@swende.se>

karalabe committed:
cmd: add state.scheme to the database flag group for local flag handling (#28107)

karalabe committed:
core, params: update Holesky testnet to relaunched spec (#28191)

karalabe committed:
params: update hash for Holesky relaunch (#28192)

msmania committed:
core/bloombits: fix deadlock when matcher session hits an error (#28184)

When MatcherSession encounters an error, it attempts to close the session.
Closing waits for all goroutines to finish, including the 'distributor'. However, the
distributor will not exit until all requests have returned.

This patch fixes the issue by delivering the (empty) result to the distributor
before calling Close().

MariusVanDerWijden committed:
core/forkid: add forkid test for holesky (#28193)

fjl committed:
internal/debug: add --log.rotate to the logging category (#28190)

jsvisa committed:
cmd/geth: ensure db is closed before exit (#28150)

buddh0 committed:
cmd/utils: fix bootnodes config priority (#28095)

This fixes an issue where the --bootnodes flag was overridden by the config file.

---------

Co-authored-by: NathanBSC <Nathan.l@nodereal.io>
Co-authored-by: Felix Lange <fjl@twurst.com>

kandrianov committed:
ethclient: fix BlockReceipts parameter encoding (#28087)


Co-authored-by: Felix Lange <fjl@twurst.com>

hzysvilla committed:
core/vm: minor code formatting (#28199)

Adding a space beween function opOrigin() and opcCaller() in instruciton.go.
Adding a space beween function opkeccak256()  and opAddress() in instruciton.go.

phenix3443 committed:
eth/downloader: typo in comment (#28196)

MariusVanDerWijden committed:
eth/fetcher: allow underpriced transactions in after timeout (#28097)

This PR will allow a previously underpriced transaction back in after a timeout
of 5 minutes. This will block most transaction spam but allow for transactions to
be re-broadcasted on networks with less transaction flow.

---------

Co-authored-by: Felix Lange <fjl@twurst.com>

s1na committed:
internal/ethapi: eth_call block parameter is optional (#28165)

So apparently in the spec the base block parameter of eth_call is optional.
I agree that "latest" is a sane default for this that most people would use.

rjl493456442 committed:
eth/downloader: remove header rollback mechanism (#28147)

* eth/downloader: remove rollback mechanism in downloader

* eth/downloader: remove the tests

bnovil committed:
core/txpool: fix typos (#28208)

core/txpool:fix typos

bstnxbt committed:
core/txpool: fix typos (#28213)

fix(core/txpool): fix typos

rjl493456442 committed:
core, accounts, eth, trie: handle genesis state missing (#28171)

* core, accounts, eth, trie: handle genesis state missing

* core, eth, trie: polish

* core: manage txpool subscription in mainpool

* eth/backend: fix test

* cmd, eth: fix test

* core/rawdb, trie/triedb/pathdb: address comments

* eth, trie: address comments

* eth: inline the function

* eth: use synced flag

* core/txpool: revert changes in txpool

* core, eth, trie: rename functions

karalabe committed:
params: release Geth v1.13.2

karalabe committed:
params: begin v1.13.3 release cycle

lightclient committed:
params: update 4788 beacon roots contract addr (#28205)

This change contains the final (?) address for 4788 beacon root contract. The update to the EIP is being tracked here: https://github.com/ethereum/EIPs/pull/7672

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

jsvisa committed:
internal/ethapi: compact db missing key starts with 0xff (#28207)

Signed-off-by: jsvisa <delweng@gmail.com>

phenix3443 committed:
core, eth: typos and some code formatting (#28201)

* fix: typo

* feat: revert symbol name

karalabe committed:
ethdb, internal/ethapi: support exposing Pebble stats too, beside LevelDB (#28224)

ethdb, internal/ethapi: support exposing Pebble stats too, besinde LevelDB

bstnxbt committed:
core: fix typos (#28218)

* fix(core/txpool): fix typos

* core/asm: fix typos

* core/bloombits: fix typos

* core/rawdb: fix typos

s1na committed:
core: infer blobGasUsed in chain maker (#28212)

Same way that the gasUsed in header is updated when a tx 
is added we should update blob gas used instead of requiring caller
to set it manually.

aaronbuchwald committed:
core/state: small trie prefetcher nits (#28183)

Small trie prefetcher nits

karalabe committed:
common: remove address.Hash footgun (#28228)

rjl493456442 committed:
ethdb/pebble: upgrade pebble to master (aa077af62593) (#28070)

* ethdb/pebble: upgrade pebble

* ethdb/pebble, go.mod: update pebble to master (aa077af62593)

---------

Co-authored-by: Péter Szilágyi <peterke@gmail.com>

karalabe committed:
ethdb/pebble: luv you linter

lightclient committed:
eth/catalyst: add validation error in new paylaod hash mismatch (#28226)

* eth/catalyst: add validation error in new paylaod hash mismatch

* eth/catalyst/api: refactor api.invalid(..) to return nil latest valid hash if none provided

karalabe committed:
cmd, eth: switch the dev synctarget to hash from block (#28209)

* cmd, eth: switch the dev synctarget to hash from block

* cmd/utils, eth/catalyst: terminate node wyen synctarget reached

MariusVanDerWijden committed:
core: implement BLOBBASEFEE opcode (0x4a) (#28098)

Implements "EIP-7516: BLOBBASEFEE opcode" for cancun, as per spec: https://eips.ethereum.org/EIPS/eip-7516

tylerni7 committed:
eth, rpc: add configurable option for wsMessageSizeLimit (#27801)

This change adds a configurable limit to websocket message. 
---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

lightclient committed:
cmd/evm: cancun-updates for b11r and t8n -tools (#28195)

This change updates `evm b11r` (blockbuilder) and `evm t8n` (transition) tools to contain cancun updates (e.g. new header fields)
---------

Co-authored-by: Mario Vega <marioevz@gmail.com>

bstnxbt committed:
core: fix typos (#28238)

DeVil2O committed:
trie: fix benchmark by ensuring key immutability (#28221)

This change fixes the bug in a benchmark, where the input to the trie is reused in a way which is not correct. 

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

hyunchel committed:
rpc: fix erroneous error-message in test (#28227)

karalabe committed:
cmd/devp2p, eth: drop eth/66 (#28239)

* cmd/devp2p, eth: drop eth/66

* eth/protocols/eth: yes sir, linter

karalabe committed:
core, eth, miner: start propagating and consuming blob txs (#28243)

* core, eth, miner: start propagating and consuming blob txs

* eth/protocols/eth: disable eth/67 if Cancun is enabled

* core/txpool, eth, miner: pass gas limit infos in lazy tx for mienr filtering

* core/txpool, miner: add lazy resolver for pending txs too

* core, eth: fix review noticed bugs

* eth, miner: minor polishes in the mining and announcing logs

* core/expool: unsubscribe the event scope

karalabe committed:
eth: when snap is complaining for missing eth, be verbose about the details (#28249)

* eth: when snap is complaining for missing eth, be verbost about the details

* eth: lower snapshot registration error verbosity

tactical-retreat committed:
cmd/evm/internal/t8ntoo: tiny bugfix for difficulty field (#28245)

hyunchel committed:
core/txpool/legacypool: fix typo (#28258)

holiman committed:
trie: refactor stacktrie (#28233)

This change refactors stacktrie to separate the stacktrie itself from the
internal representation of nodes: a stacktrie is not a recursive structure
of stacktries, rather, a framework for representing and operating upon a set of nodes.

---------

Co-authored-by: Gary Rong <garyrong0905@gmail.com>

bstnxbt committed:
core: fix typos (#28255)

fixes various typos in core

hyunchel committed:
eth, params: fix typos (#28286)

* eth/ethconfig: fix typo on comment

* params/config: fix typo on comment

* eth/ethconfig: fix typo on comment

holiman committed:
all: move light.NodeSet to trienode.ProofSet (#28287)

This is a minor refactor in preparation of changes to range verifier. This PR contains no intentional functional changes but moves (and renames) the light.NodeSet

bnovil committed:
trie: fix a typo, use correct docstrings (#28302)

* fix a typo

* trie: additional fixes to docstrings

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

karalabe committed:
eth: enforce announcement metadatas and drop peers violating the protocol (#28261)

* eth: enforce announcement metadatas and drop peers violating the protocol

* eth/fetcher: relax eth/68 validation a bit for flakey clients

* tests/fuzzers/txfetcher: pull in suggestion from Marius

* eth/fetcher: add tests for peer dropping

* eth/fetcher: linter linter linter linter linter

lightclient committed:
git: ignore tests/spec-tests folder (#28254)

karalabe committed:
accounts/abi/bind/backend: use requested header for gas prices and gas limits (#28280)

holiman committed:
trie: remove owner and binary marshaling from stacktrie (#28291)

This change
  - Removes the owner-notion from a stacktrie; the owner is only ever needed for comitting to the database, but the commit-function, the `writeFn` is provided by the caller, so the caller can just set the owner into the `writeFn` instead of having it passed through the stacktrie.
  - Removes the `encoding.BinaryMarshaler`/`encoding.BinaryUnmarshaler` interface from stacktrie. We're not using it, and it is doubtful whether anyone downstream is either.

rjl493456442 committed:
ethdb/pebble: add level file metrics (#28271)

karalabe committed:
eth/fetcher: throttle tx fetches to 128KB responses (#28304)

* eth/fetcher: throttle tx fetches to 128KB responses

* eth/fetcher: unindent a clause per review request

karalabe committed:
eth/fetcher: fix typo

rjl493456442 committed:
cmd, core: resolve scheme from a read-write database (#28313)

* cmd, core: resolve scheme from a read-write database

* cmd, core, eth: move the scheme check in the ethereum constructor

* cmd/geth: dump should in ro mode

* cmd: reverts

vuittont60 committed:
accounts, cmd: fix typos (#28300)

holiman committed:
cmd/geth: fix failing test (#28322)

karalabe committed:
params: release Geth v.1.13.3

karalabe committed:
params: begin v1.13.4 release cycle

rjl493456442 committed:
eth/protocols/snap: fix snap sync failure on empty storage range (#28306)

This change addresses an issue in snap sync, specifically when the entire sync process can be halted due to an encountered empty storage range.

Currently, on the snap sync client side, the response to an empty (partial) storage range is discarded as a non-delivery. However, this response can be a valid response, when the particular range requested does not contain any slots.

For instance, consider a large contract where the entire key space is divided into 16 chunks, and there are no available slots in the last chunk [0xf] -> [end]. When the node receives a request for this particular range, the response includes:

    The proof with origin [0xf]
    A nil storage slot set

If we simply discard this response, the finalization of the last range will be skipped, halting the entire sync process indefinitely. The test case TestSyncWithUnevenStorage can reproduce the scenario described above.

In addition, this change also defines the common variables MaxAddress and MaxHash.

karalabe committed:
go.mod, internal/build: update Azure dependencies (#28329)

karalabe committed:
go.mod: update gnark lib to fix a malleability issue (#28333)

karalabe committed:
go.mod: update fastcache to latest (#28334)

holiman committed:
build: move version-info into checksum file (#28324)

* build: upgrade to golang 1.21.2

* build: verify checksums via tool

* deps: upgrade go to 1.21.3

* build: move more build metadata into checksum file

* build: move gobootsrc to checksums

karalabe committed:
go.mod: update AWS APIs to latest heads (#28332)

holiman committed:
trie: make rhs-proof align with last key in range proofs (#28311)

During snap-sync, we request ranges of values: either a range of accounts or a range of storage values. For any large trie, e.g. the main account trie or a large storage trie, we cannot fetch everything at once.

Short version; we split it up and request in multiple stages. To do so, we use an origin field, to say "Give me all storage key/values where key > 0x20000000000000000". When the server fulfils this, the server provides the first key after origin, let's say 0x2e030000000000000 -- never providing the exact origin. However, the client-side needs to be able to verify that the 0x2e03.. indeed is the first one after 0x2000.., and therefore the attached proof concerns the origin, not the first key.

So, short-short version: the left-hand side of the proof relates to the origin, and is free-standing from the first leaf.

On the other hand, (pun intended), the right-hand side, there's no such 'gap' between "along what path does the proof walk" and the last provided leaf. The proof must prove the last element (unless there are no elements).

Therefore, we can simplify the semantics for trie.VerifyRangeProof by removing an argument. This doesn't make much difference in practice, but makes it so that we can remove some tests. The reason I am raising this is that the upcoming stacktrie-based verifier does not support such fancy features as standalone right-hand borders.

karalabe committed:
go.mod: pull in the latest cloudflare API libs (#28336)

karalabe committed:
cmd, core, ethdb: enable Pebble on 32 bits and OpenBSD too (#28335)

* cmd, core, ethdb: enable Pebble on 32 bits and OpenBSD too

* ethdb/pebble: use Pebble's internal constant calculation

meowsbits committed:
Merge branch 'master' into merge/foundation-release/1.13.1

 Conflicts:
	go.mod
	go.sum

meowsbits committed:
go.mod,go.sum: go mod tidy

Date: 2023-10-16 09:50:51-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge branch 'release/1.13' into merge/foundation-release/1.13.3

 Conflicts:
	.gitignore
	accounts/abi/bind/backends/simulated.go
	cmd/evm/internal/t8ntool/execution.go
	cmd/evm/main.go
	cmd/evm/runner.go
	cmd/evm/staterunner.go
	cmd/utils/flags.go
	core/forkid/forkid_test.go
	core/genesis.go
	core/genesis_test.go
	core/state_processor.go
	core/vm/gas_table.go
	core/vm/jump_table.go
	eth/downloader/downloader.go
	eth/downloader/downloader_test.go
	eth/downloader/peer.go
	eth/ethconfig/config.go
	eth/handler.go
	eth/protocols/eth/handler.go
	eth/protocols/eth/handler_test.go
	eth/sync.go
	go.mod
	go.sum
	internal/ethapi/api_test.go
	les/client_handler.go
	miner/miner_test.go
	miner/worker.go
	params/config.go
	params/version.go

meowsbits committed:
cmd/evm,cmd/utils,core,core/vm,eth/ethconfig,eth,go.sum: auto-optimize imports, go mod

Date: 2023-10-16 10:29:29-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
go.mod,go.sum: go mod tidy

Date: 2023-10-16 10:29:45-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/coregeth,params/types/ctypes,params/types/genesisT,params/types/goethereum: config.GetEIP7516TransitionTime undefined

Date: 2023-10-16 10:41:23-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: "github.com/ethereum/go-ethereum/consensus/misc" imported and not used

Date: 2023-10-16 10:42:39-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/evm/internal/t8ntool: undefined: params (typecheck)

Date: 2023-10-16 10:43:09-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/evm/internal/t8ntool: chainConfig.ChainID undefined

Date: 2023-10-16 10:43:40-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/downloader: not enough arguments in call to d.processHeaders

Date: 2023-10-16 10:52:55-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/downloader: assignment mismatch: 3 variables but p.peer.Head returns 2 values

Date: 2023-10-16 10:53:53-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Revert "eth/downloader: remove header rollback mechanism (#28147)"

This reverts commit b85c183ea7417e973dbbccd27b3fb7d7097b87dd.

 Conflicts:
	eth/downloader/downloader.go

holiman committed:
tests: update execution-spec-tests to 1.0.5 (#28337)

Updates execution-spec-tests to 1.0.5: https://github.com/ethereum/execution-spec-tests/releases/tag/v1.0.5, switching to develop which contains Cancun tests (which are also enabled in this change).

meowsbits committed:
eth/downloader: fetchTotalDifficulty method comment

Date: 2023-10-16 12:15:02-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Revert "eth/downloader: assignment mismatch: 3 variables but p.peer.Head returns 2 values"

This reverts commit c4f987f24a04ab93a79f2ac53eac4f54bf2f6bd7.

meowsbits committed:
eth/downloader: p.peer.SetHead undefined (type Peer has no field or method SetHead))

Date: 2023-10-16 12:34:41-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
miner: undefined: params

Date: 2023-10-16 12:36:04-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
miner: invalid operation: cannot call non-function ltx.Hash (variable of type "github.com/ethereum/go-ethereum/common".Hash))

Date: 2023-10-16 12:38:23-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth: not enough arguments in call to downloader.New

Date: 2023-10-16 13:00:38-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth: cannot convert *res.Res.(*eth.BlockHeadersPacket)

Date: 2023-10-16 13:02:25-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth:  h.acceptTxs undefined

Date: 2023-10-16 13:03:57-06:00
Signed-off-by: meows <b5c6@protonmail.com>

karalabe committed:
eth/fetcher: only make noise big mismatches (#28356)

phenix3443 committed:
fix: broken url link (#28342)

holiman committed:
internal/ethapi: fix codehash lookup in eth_getProof (#28357)

This change fixes #28355, where eth_getProof failed to return the correct codehash under certain conditions. This PR changes the logic to unconditionally look up the codehash, and also adds some more tests.

xiaolou86 committed:
cmd: fix typos (#28323)

MariusVanDerWijden committed:
eth/fetcher: fix fetcher timeout (#28220)

This changes fixes a bug in the fetcher, where the timeout for how long to remember underpriced transaction was erroneously compared, and the timeout never hit.
---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

karalabe committed:
params: release Geth v1.13.4

meowsbits committed:
cmd/evm: undefined: core.Genesis

Date: 2023-10-17 06:53:45-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/forkid: undefined: core.DefaultHoleskyGenesisBlock

Date: 2023-10-17 07:07:39-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/evm: genesisConfig.MustCommit undefined

Date: 2023-10-17 07:11:11-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth: undefined: eth.ETH66

Note that this test function is uniq
to core-geth now.
The test passes with ETH67 and GetBlockHeadersRequest
typing.

Date: 2023-10-17 07:16:47-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/downloader: undefined: eth.ETH66

I removed the ETH66 tests; ETH67 exist;
and added the ETH68 complement.

Date: 2023-10-17 07:20:41-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/fetcher: undefined: params.MaxBlobGasPerBlock

Date: 2023-10-17 07:21:52-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les/downloader: undefined: eth.ETH66

Date: 2023-10-17 07:22:24-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les/downloader: undefined: eth.NodeDataMsg (rm -rf les/downloader)

Its gone upstream.

Date: 2023-10-17 07:29:48-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
miner: undefined: params.ChainConfig

Date: 2023-10-17 07:30:30-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: undefined: eth.ETH66

A great many changes in this file,
so its now identical to upstream.
See if that works.
Light syncing might be dead btw.

Date: 2023-10-17 07:34:25-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: b.eth.handler.downloader undefined

Date: 2023-10-17 07:42:47-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: rm sync files

Date: 2023-10-17 07:45:12-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge branch 'release/1.13' into merge/foundation-release/1.13.4

 Conflicts:
	ethclient/gethclient/gethclient_test.go
	go.mod
	go.sum
	params/version.go

meowsbits committed:
go.mod,go.sum,les: go mod tidy

Date: 2023-10-17 07:51:55-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: fix typo

Date: 2023-10-17 07:52:56-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les/catalyst,les/checkpointoracle,les/fetcher: tons of errors in less

So I rm -rfed all packages that only core-geth
had. go-ethereum has gone ham removing stuff
they dont need.

Date: 2023-10-17 07:56:51-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: undefined types; move to match go-ethereum

Date: 2023-10-17 07:59:17-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: b.eth.Downloader undefined

Date: 2023-10-17 08:00:14-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: c.handler.fetcher undefined

Date: 2023-10-17 08:00:55-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: could not load export data: no export data for "github.com/ethereum/go-ethereum/les/checkpointoracle"

Date: 2023-10-17 08:02:20-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: remove oracle

Date: 2023-10-17 08:03:44-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: api.backend.oracle undefined

Date: 2023-10-17 08:05:48-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: server.oracle undefined

Date: 2023-10-17 08:06:08-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: remove more checkpoint and oracle logic

Date: 2023-10-17 08:10:56-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: unknown field oracle in struct literal of type lesCommons

Date: 2023-10-17 08:12:51-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: unnecessary conversion (unconvert)

Date: 2023-10-17 08:14:20-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: var `errNoCheckpoint` is unused

Date: 2023-10-17 08:14:37-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: func `(*pruner).close` is unused

Date: 2023-10-17 08:15:24-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: undefined: pruner

Date: 2023-10-17 08:15:50-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: var `oracleAddr` is unused

Date: 2023-10-17 08:17:00-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: undefined: oracleAddr

Date: 2023-10-17 08:17:25-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: unused: var auth (no deploy checkpoint contract)

Date: 2023-10-17 08:18:22-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Revert "les: remove obsolete code related to PoW header syncing (#27737)"

This reverts commit d4d88f9bce13ca9310bf28f5f26ea9f1915ba90d.

 Conflicts:
	eth/ethconfig/gen_config.go
	les/api_test.go
	les/client.go

meowsbits committed:
eth/ethconfig: problem: duplicate fields

Date: 2023-10-17 09:31:50-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/protocols/eth: maintain ETH66

Date: 2023-10-17 09:32:18-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les/catalyst,les/downloader,les/fetcher,les: undefined: params.XXX -> vars.XXX

Date: 2023-10-17 09:34:52-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les/downloader: not enough arguments in call to chain.InsertHeaderChain

Date: 2023-10-17 09:49:31-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: not enough arguments in call to chain.InsertHeaderChain

Date: 2023-10-17 09:50:27-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les/downloader: not enough arguments in call to trie.NewDatabase

Date: 2023-10-17 09:51:31-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les/catalyst: TerminalTotalDifficulty undefined

Date: 2023-10-17 09:53:04-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les/catalyst,les/downloader,les/fetcher: undefined: core.Genesis

Date: 2023-10-17 09:54:49-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les/catalyst: not enough arguments in call to engine.ExecutableDataToBlock

Date: 2023-10-17 09:56:15-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les/catalyst: not enough arguments in call to lesService.BlockChain()

Date: 2023-10-17 09:57:05-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les/downloader,les/fetcher: gspec.MustCommit undefined

Date: 2023-10-17 09:59:34-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les/downloader: cannot use tester (variable of type *downloadTester) as BlockChain value in argument to New: *downloadTester does not implement BlockChain (wrong type for method InsertHeaderChain)

Date: 2023-10-17 10:01:04-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: remove unused vars

Date: 2023-10-17 10:04:01-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests/spec-tests/fixtures/berlin/eip2930_access_list/acl,tests/spec-tests/fixtures/cancun/eip4788_beacon_root/beacon_root_contract,tests/spec-tests/fixtures/cancun/eip4788_beacon_root/blocks_beacon_root_contract,tests/spec-tests/fixtures/cancun/eip4844_blobs/blob_txs,tests/spec-tests/fixtures/cancun/eip4844_blobs/blob_txs_full,tests/spec-tests/fixtures/cancun/eip4844_blobs/blobhash_opcode,tests/spec-tests/fixtures/cancun/eip4844_blobs/blobhash_opcode_contexts,tests/spec-tests/fixtures/cancun/eip4844_blobs/excess_blob_gas,tests/spec-tests/fixtures/cancun/eip4844_blobs/excess_blob_gas_fork_transition,tests/spec-tests/fixtures/cancun/eip4844_blobs/point_evaluation_precompile,tests/spec-tests/fixtures/cancun/eip4844_blobs/point_evaluation_precompile_gas,tests/spec-tests/fixtures/cancun/eip5656_mcopy/mcopy,tests/spec-tests/fixtures/cancun/eip5656_mcopy/mcopy_contexts,tests/spec-tests/fixtures/cancun/eip5656_mcopy/mcopy_memory_expansion,tests/spec-tests/fixtures/cancun/eip6780_selfdestruct/selfdestruct,tests/spec-tests/fixtures/frontier/opcodes/dup,tests/spec-tests/fixtures/homestead/yul/yul_example,tests/spec-tests/fixtures/istanbul/eip1344_chainid/chainid,tests/spec-tests/fixtures/merge/security/selfdestruct_balance_bug,tests/spec-tests/fixtures/shanghai/eip3651_warm_coinbase/warm_coinbase,tests/spec-tests/fixtures/shanghai/eip3855_push0/push0,tests/spec-tests/fixtures/shanghai/eip3860_initcode/initcode,tests/spec-tests/fixtures/shanghai/eip4895_withdrawals/withdrawals: i think this is what happens when you run the tests

Date: 2023-10-18 07:33:27-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/state: fix a panic in tests caused by a nil balance in gen accnt

Date: 2023-10-18 07:34:25-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params: fix holesky genesis params (first testnet failed)

Date: 2023-10-18 08:07:30-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/genesisT: problem: TestCreation fails

solution: update genesis alloc decoder
to handle new fields

Date: 2023-10-18 08:08:31-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les/downloader: re-implement frequency condition for rollback

This was missing against master; since
we maintain pow header les sync, we should
keep all validation conditions.

Date: 2023-10-19 09:03:22-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: disable checkfreq if ULC

ULC=ultra light client

Since ULC is still supported, we should keep
all related conditions for it.

Date: 2023-10-19 09:04:30-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les: re-install les API methods and service

The absence of these was causing geth
to stall out (RPC server shutting down;
silently because of No-available-methods-for-service
error)

It is reasonable to maintain these
methods because the les service should
still exist because the les feature
still exists.
HOWEVER, currently checkpointing logic
has been removed from the les service,
and all the API methods are related to
checkpointing...
so they all now return the same error
(errNotActivated).

Maybe we'll reconsider removing the checkpointing
from les, but for now
this makes the tests pass, at least.

Date: 2023-10-19 09:12:19-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/confp: guard uint64 undeflow for rewindto target

In certain test cases (execution-specs/cancun),
tests cycled infinitely, caught in the compatibility
check for-loop func Compatible.

This logic is parallel to upstream,
and they should be advised that
of the fix.

Date: 2023-10-25 17:33:44-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: reorder eip4844 in precompileds logic

I want the precompiled to be conditionally
activated in a code order
that is chronological.

Date: 2023-10-26 06:31:24-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
consensus/ethash: add EIP-4844 verification support to ethash header verification

Date: 2023-10-26 06:46:11-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
consensus/ethash: add EIP-4788 header verification

Its important to verify the negative
as well as the affirmative.

Date: 2023-10-26 06:53:40-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: differentiate 4844/4788 in Genesis block method

Date: 2023-10-26 07:22:57-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: differentiate 4844/4788 in header maker fn

Date: 2023-10-26 07:24:07-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
miner: differentiate 4844/4788 in prepareWork fn

Date: 2023-10-26 07:26:36-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: debugging: match test T encodings to release/1.13

Date: 2023-10-26 08:48:38-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests/spec-tests/fixtures/berlin/eip2930_access_list/acl,tests/spec-tests/fixtures/cancun/eip4788_beacon_root/beacon_root_contract,tests/spec-tests/fixtures/cancun/eip4788_beacon_root/blocks_beacon_root_contract,tests/spec-tests/fixtures/cancun/eip4844_blobs/blob_txs,tests/spec-tests/fixtures/cancun/eip4844_blobs/blob_txs_full,tests/spec-tests/fixtures/cancun/eip4844_blobs/blobhash_opcode,tests/spec-tests/fixtures/cancun/eip4844_blobs/blobhash_opcode_contexts,tests/spec-tests/fixtures/cancun/eip4844_blobs/excess_blob_gas,tests/spec-tests/fixtures/cancun/eip4844_blobs/excess_blob_gas_fork_transition,tests/spec-tests/fixtures/cancun/eip4844_blobs/point_evaluation_precompile,tests/spec-tests/fixtures/cancun/eip4844_blobs/point_evaluation_precompile_gas,tests/spec-tests/fixtures/cancun/eip5656_mcopy/mcopy,tests/spec-tests/fixtures/cancun/eip5656_mcopy/mcopy_contexts,tests/spec-tests/fixtures/cancun/eip5656_mcopy/mcopy_memory_expansion,tests/spec-tests/fixtures/cancun/eip6780_selfdestruct/selfdestruct,tests/spec-tests/fixtures/frontier/opcodes/dup,tests/spec-tests/fixtures/homestead/yul/yul_example,tests/spec-tests/fixtures/istanbul/eip1344_chainid/chainid,tests/spec-tests/fixtures/merge/security/selfdestruct_balance_bug,tests/spec-tests/fixtures/shanghai/eip3651_warm_coinbase/warm_coinbase,tests/spec-tests/fixtures/shanghai/eip3855_push0/push0,tests/spec-tests/fixtures/shanghai/eip3860_initcode/initcode,tests/spec-tests/fixtures/shanghai/eip4895_withdrawals/withdrawals: git checkout --no-overlay release/1.13 -- tests/spec-tests/fixtures

Date: 2023-10-26 08:50:19-06:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
core/forkid,go.mod,go.sum,params: Merge branch 'merge/foundation-release/1.13.1' into merge/foundation-release/1.13.4

diega committed:
Release/v1.12.15 (#572)

* bump version from v1.12.15-unstable to v1.12.15-stable

Signed-off-by: Diego López León <dieguitoll@gmail.com>

* bump version from v1.12.15-stable to v1.12.16-unstable

Signed-off-by: Diego López León <dieguitoll@gmail.com>

---------

Signed-off-by: Diego López León <dieguitoll@gmail.com>

meowsbits committed:
core/forkid: fix missing test field (genesis block)

Date: 2023-10-27 08:19:16-06:00
Signed-off-by: meows <b5c6@protonmail.com>

dependabot[bot] committed:
build(deps): bump github.com/docker/docker

Bumps [github.com/docker/docker](https://github.com/docker/docker) from 20.10.24+incompatible to 24.0.7+incompatible.
- [Release notes](https://github.com/docker/docker/releases)
- [Commits](https://github.com/docker/docker/compare/v20.10.24...v24.0.7)

---
updated-dependencies:
- dependency-name: github.com/docker/docker
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

meowsbits committed:
node: return an error when node.startRPC returns one

Errors returned from this function were getting
ignored, resulting in silent-but-deadly
startup errors, especially around authrpc,
which is a relatively new addition,
and the port collision of which causes
a lot of people a lot of headaches.

Date: 2023-11-01 07:51:52-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge pull request #577 from etclabscore/dependabot/go_modules/github.com/docker/docker-24.0.7incompatible

build(deps): bump github.com/docker/docker from 20.10.24+incompatible to 24.0.7+incompatible

dependabot[bot] committed:
build(deps): bump github.com/consensys/gnark-crypto

Bumps [github.com/consensys/gnark-crypto](https://github.com/consensys/gnark-crypto) from 0.10.0 to 0.12.0.
- [Release notes](https://github.com/consensys/gnark-crypto/releases)
- [Changelog](https://github.com/Consensys/gnark-crypto/blob/master/CHANGELOG.md)
- [Commits](https://github.com/consensys/gnark-crypto/compare/v0.10.0...v0.12.0)

---
updated-dependencies:
- dependency-name: github.com/consensys/gnark-crypto
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

meowsbits committed:
go.mod,go.sum: go get -u github.com/consensys/gnark-crypto

Date: 2023-11-01 08:31:57-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge pull request #578 from etclabscore/rpc-startup-error-handling

node: return an error when node.startRPC returns one

meowsbits committed:
core/forkid: implement GatherForks test for time-based forks

This should have been tested already,
but was overlooked.
Tests are passing.

Date: 2023-11-02 09:37:17-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/forkid: add GatherForks test cases for goerli, sepolia

Being default-availble networks,
they should have tests.

Date: 2023-11-02 09:40:23-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge pull request #579 from etclabscore/dependabot/go_modules/github.com/consensys/gnark-crypto-0.12.0

build(deps): bump github.com/consensys/gnark-crypto from 0.10.0 to 0.12.0

dependabot[bot] committed:
build(deps): bump golang.org/x/image from 0.7.0 to 0.10.0

Bumps [golang.org/x/image](https://github.com/golang/image) from 0.7.0 to 0.10.0.
- [Commits](https://github.com/golang/image/compare/v0.7.0...v0.10.0)

---
updated-dependencies:
- dependency-name: golang.org/x/image
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

meowsbits committed:
Merge pull request #581 from etclabscore/dependabot/go_modules/golang.org/x/image-0.10.0

build(deps): bump golang.org/x/image from 0.7.0 to 0.10.0

meowsbits committed:
Merge pull request #580 from etclabscore/forkid-tests

Forkid tests

meowsbits committed:
bump version from v1.12.16-unstable to v1.12.16-stable

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
bump version from v1.12.16-stable to v1.12.17-unstable

Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr commented on core/vm/jump_table.go:
```diff
@@ -211,8 +211,22 @@ func instructionSetForConfig(config ctypes.ChainConfigurator, isPostMerge bool,
+			return validate(instructionSet)
+		}
+	*/
 	if config.IsEnabledByTime(config.GetEIP4844TransitionTime, bt) {
```
Shall we already add enablement using block number for ETC? Or we prefer doing this before hard-forks for needed EIPs only?

meowsbits replied:
I'd say its a TODO. It seems unfinished without them.

ziogaschr replied:
Done with [de0e288](https://github.com/etclabscore/core-geth/pull/575/commits/de0e2883e9d6e3af1a56fb239d5c40ee95f6e8b4)

meowsbits committed:
Merge branch 'master' into merge/foundation-release/1.13.4

Conflicts:
      core/forkid/forkid_test.go
      go.mod
      go.sum

ziogaschr commented on eth/downloader/downloader.go:
```diff
@@ -1363,7 +1360,7 @@ func (d *Downloader) fetchReceipts(from uint64, beaconMode bool) error {
 // processHeaders takes batches of retrieved headers from an input channel and
 // keeps processing and scheduling them into the header chain and downloader's
 // queue until the stream ends or a failure occurs.
-func (d *Downloader) processHeaders(origin uint64, ttd *big.Int, beaconMode bool) error {
```
Seems that merge bring some code changes that we don't need with regards https://github.com/etclabscore/core-geth/commit/b2a65e4dd018dbf5d7c25ba94affa8e68d482860#diff-80cc409bf7f1e4a3d19174ed878e4dcb3d40b19b6e0580bd3816a174c57e0fb2

meowsbits replied:
You mean just the new unused `td` variable in the signature? IMO it's ok to have that, since the linter doesn't complain about it and apparently it'll make merge conflicts one line fewer. The rest of the Zurich logic looks in place to me.

Re: Zurich; See also `merge/foundation-release/1.13.5` (https://github.com/etclabscore/core-geth/pull/593), in case they or I fixed something in this version in the later one.

ziogaschr commented on eth/protocols/eth/handler.go:
```diff
@@ -100,10 +95,12 @@ func MakeProtocols(backend Backend, network uint64, protocolVersions []uint, dns
-		if version <= ETH66 && backend.Chain().TrieDB().Scheme() == rawdb.PathScheme {
+		// TODO(meowsbits): FIXME re: Cancun config/time/enabled check
+		// Blob transactions require eth/68 announcements, disable everything else
+		if version <= ETH67 && backend.Chain().Config().GetEIP4844TransitionTime() != nil {
```
@ziogaschr: check if `version <= ETH67` change is intentional

ziogaschr replied:
It is, as ETH66 got dropped

ziogaschr committed:
cmd/evm/internal/t8ntool,consensus/beacon,consensus/clique,consensus/ethash,core,core/txpool/blobpool,core/txpool,core/types,core/vm,eth/catalyst,eth/protocols/eth,miner,params/types/coregeth,params/types/ctypes,params/types/genesisT,params/types/goethereum: add conditions per block number for cancun EIPs

ziogaschr commented on eth/catalyst/api.go:
```diff
@@ -218,14 +218,14 @@ func (api *ConsensusAPI) verifyPayloadAttributes(attr *engine.PayloadAttributes)
 	// Verify withdrawals attribute for Shanghai.
 	withdrawalsCheck := func(b *big.Int, t uint64) bool {
-		return c.IsEnabledByTime(c.GetEIP4895TransitionTime, &t)
+		return c.IsEnabledByTime(c.GetEIP4895TransitionTime, &t) || c.IsEnabled(c.GetEIP4895Transition, b)
```
this wasn't in place from the last merge and I added it now @meowsbits. Any reason not to do it?

meowsbits replied:
I don't think there's any reason not -- thanks for the catch.

ziogaschr commented on core/txpool/blobpool/blobpool_test.go:
```diff
@@ -102,8 +102,9 @@ func (bc *testBlockChain) CurrentBlock() *types.Header {
 		blockNumber = new(big.Int).Add(new(big.Int).SetUint64(*bc.config.GetEIP1559Transition()), big.NewInt(1))
-		blockTime   = *bc.config.GetEIP4844TransitionTime() + 1
-		gasLimit    = uint64(30_000_000)
+		// TODO @ziogaschr: consider what to put here using the block number. We can leave it as is, as it's been using in tests only
```
@meowsbits @diega what do you think of this one?

meowsbits replied:
I think its probably fine to leave until we/anybody else wants to use EIP4844/blobpools in their chain config. The test also assumes `EIP1559` which would potentially need refactoring too.

ziogaschr replied:
Was thinking to use something like this, but I don't see this code being tested with blockNumbers if we are not going to use this, so let's not add more unused code to maintain.

```
	// If EIP4844 is enabled using block number and it's bigger than EIP1559
	// activation block then use EIP4844's block number
	if eip4844BlockNumber := bc.config.GetEIP4844Transition(); eip4844BlockNumber != nil {
		if blockNumber.Cmp(new(big.Int).SetUint64(*eip4844BlockNumber)) == -1 {
			blockNumber = new(big.Int).SetUint64(*eip4844BlockNumber)
		}
	}
```

ziogaschr committed:
core/forkid: forkid_test.go:563:40: not enough arguments in call to gatherForks

forkid_test.go:563:40: not enough arguments in call to gatherForks
    have (ctypes.ChainConfigurator)
    want (ctypes.ChainConfigurator, uint64) (typecheck)

ziogaschr committed:
core/txpool/blobpool: 🧹

diega committed:
Bump go version for linting GHA (#618)

Signed-off-by: Diego López León <dieguitoll@gmail.com>

ziogaschr committed:
params/types/goethereum: fix go_ethereum configurator for Cancun by block number
